### PR TITLE
apple: paste images in editor

### LINKED ISSUE #2113: pasting images in apple editor
Pasted images in the lockbook editor should be automatically imported and insert the appropriate markdown syntax for the image to appear.

### PR DESCRIPTION
**Pull request by smailbarkouch**
Images can now be pasted into the editor. Lockbook will automatically import the file and insert the appropriate markdown syntax for the image to appear.

Demonstration
<details>

https://github.com/lockbook/lockbook/assets/20663038/bbdce5d5-9600-45d0-92d8-ed0b8d33244d
</details>

fixes #2113

### CODE REVIEW

**Review by Parth — CHANGES_REQUESTED** [review]
paste in markdown image format

File: clients/apple/Shared/Stateful Logic/DocumentService.swift [hunk 1]
```diff
@@ -282,7 +282,16 @@ class DocumentLoadingInfo: ObservableObject {
             DispatchQueue.main.async {
                 switch operation {
                 case .success(let txt):
-                    self.textDocument = EditorState(text: txt, isiPhone: self.isiPhone)
+                    self.textDocument = EditorState(text: txt, isiPhone: self.isiPhone) { url in
+                        DI.importExport.importFilesSync(sources:[url.path(percentEncoded: false)], destination: self.meta.parent)
+                        
+                        if let parentPath = DI.files.getPathByIdOrParent(maybeId: self.meta.parent) {
+                            
+                            return "lb:/\(parentPath)\(url.lastPathComponent)"
```
Comment (tvanderstad) [inline clients/apple/Shared/Stateful Logic/DocumentService.swift:290]: two slashes?

Comment (tvanderstad) [reply clients/apple/Shared/Stateful Logic/DocumentService.swift:290]: not sure where exactly this belongs but looking for something like this
```suggestion
                            return "![alt text](lb://\(parentPath)\(url.lastPathComponent))"
```

File: clients/apple/Shared/Stateful Logic/DocumentService.swift [hunk 2]
```diff
@@ -282,7 +284,17 @@ class DocumentLoadingInfo: ObservableObject {
             DispatchQueue.main.async {
                 switch operation {
                 case .success(let txt):
-                    self.textDocument = EditorState(text: txt, isiPhone: self.isiPhone)
+                    self.textDocument = EditorState(text: txt, isiPhone: self.isiPhone) { url in
+                        DI.importExport.importFilesSync(sources:[url.path(percentEncoded: false)], destination: self.meta.parent)
+                        
+                        if let parentPath = DI.files.getPathByIdOrParent(maybeId: self.meta.parent) {
+                            if let file = DI.files.getFileByPath(path: parentPath + url.lastPathComponent) {
+                                return "![alt text](lb://\(file.id.uuidString.lowercased()))"
```
Comment (Parth) [inline clients/apple/Shared/Stateful Logic/DocumentService.swift:292]: I wonder if we could have better alt text

**Review by Parth — APPROVED** [review]

File: clients/apple/Shared/Stateful Logic/DocumentService.swift (hunk 2 above)
Comment (smailbarkouch) [reply clients/apple/Shared/Stateful Logic/DocumentService.swift]: I changed the alt name to the image's name (minus the extension).

File: clients/apple/Shared/Stateful Logic/DocumentService.swift (hunk 1 above)
Comment (smailbarkouch) [reply clients/apple/Shared/Stateful Logic/DocumentService.swift]: You're right, changed it to the image markdown syntax.

**Pull request merged**